presto-product-tests/src/main/java/com/facebook/presto/tests/IntersectTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests;
+
+import com.facebook.presto.tests.ImmutableTpchTablesRequirements.ImmutableNationTable;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.Requires;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.tests.TestGroups.INTERSECT;
+import static com.facebook.presto.tests.TestGroups.QUERY_ENGINE;
+import static com.teradata.tempto.assertions.QueryAssert.Row.row;
+import static com.teradata.tempto.assertions.QueryAssert.assertThat;
+import static com.teradata.tempto.query.QueryExecutor.query;
+import static com.teradata.tempto.query.QueryType.SELECT;
+
+@Requires(ImmutableNationTable.class)
+public class IntersectTests
```

Comment:
put all these queries to AbstractQueriesTest. Leave one smoke query as product tests.
AbstractQueriesTest are used for all the connectors, not only hive.